c/driver/postgresql/connection.cc

```diff
@@ -126,6 +126,181 @@ class PqResultHelper {
   std::string query_;
 };
 
+class PqGetObjectsHelper {
+ public:
+  PqGetObjectsHelper(PGconn* conn, int depth, const char* catalog, const char* db_schema,
+                     const char* table_name, const char** table_types,
+                     const char* column_name, struct ArrowSchema* schema,
+                     struct ArrowArray* array, struct AdbcError* error)
+      : conn_(conn),
+        depth_(depth),
+        catalog_(catalog),
+        db_schema_(db_schema),
+        table_name_(table_name),
+        table_types_(table_types),
+        column_name_(column_name),
+        schema_(schema),
+        array_(array),
+        error_(error) {}
+
+  AdbcStatusCode GetObjects() {
+    RAISE_ADBC(AdbcInitConnectionObjectsSchema(schema_, error_));
```

Comment:
Not sure if this should stay here, move to the init with an abort, or even be contained within the class

Reply:
We should avoid aborts if possible; I think it can stay here (if it's only used here, it makes sense as a method IMO)